lib/client/node.js

```diff
@@ -291,7 +291,7 @@ class NodeClient extends Client {
 
   estimateFee(blocks) {
     assert(blocks == null || typeof blocks === 'number');
-    return this.call('estimate fee', blocks);
+    return this.get(`/fee?blocks=${blocks}`);
```

Comment:
So the RPC returns `Amount.btc(fee, true)` and the HTTP endpoint returns a different result with `{ rate: fee }`.

Reply:
Hmm, yes. There are numerous notes throughout api-docs about satoshis/BTC in the fee rate arguments: https://bcoin.io/api-docs/#send-a-transaction

Reply:
Yeah, generally I think satoshis is preferred. This would be a case that the CLI gives it in satoshis (contrary to that comment).

Without this addition, I'm currently getting nothing back from the result.

Reply:
> Without this addition, I'm currently getting nothing back from the result (it may be silently failing).

You mean using the direct JS method right? `NodeClient.estimateFee()`? That method makes a socket call to the node/http module. And there *IS* a hook for it:

https://github.com/bcoin-org/bcoin/blob/master/lib/node/http.js#L495-L503

...but it won't be "hook'ed" because the socket has not been authorized. My first draft of this PR was adding `this.client.open()` in `bcoin-cli` before calling `estimateFee()`. And this works, but it would be the only CLI command that made a socket call, it seemed out of style. However, `bcoin-cli.destroy()` does have a `this.client.close()` already, even though it will never be called -- perhaps a remnant of an older version of the CLI tool?

Meanwhile, the `cURL` method always works because the http endpoint is set up like the rest of the API. So for this PR I decided to scrap the socket auth and use the RESTful style.

Reply:
Okay, taking a look at this again. And it looks like the previous result was in satoshis or btc for `client.estimateFee()`. This should probably noted in the changelog? I'm using regtest for testing and the rate is `0`, so haven't verified what unit was previous.